IndicoToolkit/Association/Positioning.cs

```diff
@@ -0,0 +1,321 @@
+            {
+                isAbove = true;
+                float overlapAmount = getHorizontalOverlap(abovePos, belowPos);
+                if (minOverlapPercent != null && overlapAmount < minOverlapPercent)
+                {
+                    isMinOverlap = false;
+                }
+            }
+            return isAbove && isMinOverlap;
+        }
+
+        /// <summary>
+        /// Check if two box positions are located on the same level/row, i.e. yaxes overlap.
+        /// </summary>
+        /// <param name="pos1">First position</param>
+        /// <param name="pos2">Second position</param>
+        /// <param name="mustBeSamePage">Required to be on same page. Defaults to true.</param>
+        public bool positionedOnSameLevel(Position pos1, Position pos2, bool mustBeSamePage = true)
+        {
+            bool sameLevel = false;
+            if (mustBeSamePage && !onSamePage(pos1, pos2))
+            {
+                sameLevel = false;
+            }
+            else if (yAxisOverlap(pos1, pos2))
+            {
+                sameLevel = true;
+            }
+            return sameLevel;
+        }
+
+        /// <summary>
+        /// Get the minimum distance between any two corners of two bounding boxes via the pythagorean formula.
+        /// </summary>
+        /// <param name="pos1">First position</param>
+        /// <param name="pos2">Second position</param>
+        /// <param name="pageHeight">If you want to measure distance across pages, set the OCR page height otherwise locations on separate pages will raise an exception. Defaults to -1.</param>
+        public float getMinDistance(Position pos1, Position pos2, int? pageHeight = null)
+        {
+            bool addPageHeight = false;
+            int pageDifference = Math.Abs(pos1.pageNum - pos2.pageNum);
+            if (pageDifference > 0)
+            {
+                if (pageHeight == null)
+                {
+                    throw new ToolkitInputException(
+                        "Predictions are not on the same page! Must enter a page height"
+                    );
+                }
+                else
+                {
+                    addPageHeight = true;
+                }
+            }
+            List<float> distances = new List<float>();
+            List<Tuple<string, string>> corners = new List<Tuple<string, string>>() {
+                new Tuple<string, string>("bbRight", "bbTop"),
+                new Tuple<string, string>("bbRight", "bbBot"),
+                new Tuple<string, string>("bbLeft", "bbTop"),
+                new Tuple<string, string>("bbLeft", "bbBot"),
+            };
+            foreach (Tuple<string, string> p1 in corners)
+            {
+                foreach (Tuple<string ,string> p2 in corners)
+                {
+                    float distance = distanceBetweenPoints(
+                        pos1.getCorner(p1.Item1, p1.Item2),
```

Comment:
would like to hear how this works

Reply:
Correct me if I'm wrong, but I believe the function compares all the distances between all the corners for a total of 16 comparisons:

pos1 (R, T) - pos2 (R, T)
pos1 (R, T) - pos2 (R, B)
pos1 (R, T) - pos2 (L, T)
pos1 (R, T) - pos2 (L, B)
pos1 (R, B) - pos2(R, T)
pos1 (R, B) - pos2 (R, B)
...
...